### PR TITLE
Update Project.toml

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -25,21 +25,21 @@ SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
...
 LinearOperators = "2.3.2"
 MathOptInterface = "1.13.2"
-NLPModels = "0.19.2"
-NLPModelsJuMP = "0.11.1"
-NLPModelsModifiers = "0.6.3"
+NLPModels = "0.21.2"
+NLPModelsJuMP = "0.13.2"
+NLPModelsModifiers = "0.7.2"
 PartitionedStructures = "0.1.5"
 PartitionedVectors = "0.1.2"
```
Comment (amontoison) [inline Project.toml:40]:
```suggestion
PartitionedVectors = "0.1.3"
```

**Review by tmigot — APPROVED** [review]
Thanks @amontoison ! This is a breaking change.

**Comment by amontoison** [comment]
@tmigot I plan to require Julia 1.10 for the future major release of `ADNLPModels.jl` (easier for the support of DI.jl and Enzyme.jl).
I propose to also require Julia 1.10 for packages that depend on it in the organization if we do a major release.
What do you think?

**Comment by tmigot** [comment]
In general, I agree we should start updating the whole JSO to 1.10.

**Pull request merged**